tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -0,0 +1,1414 @@
+   RTFStreamerInfoList streamerInfo;
+   auto classTagOffset = keyStreamerInfo.fKeyLen +
+      offsetof(struct RTFStreamerInfoList, fStreamerInfo) +
+      offsetof(struct RTFStreamerInfoObject, fStreamers) +
+      offsetof(struct RTFStreamerVersion, fNewClassTag) + 2;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerSize.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerSeekHeader.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerNBytesHeader.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerLenHeader.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerSeekFooter.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerNBytesFooter.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerLenFooter.fClassTag = 0x80000000 | classTagOffset;
+   streamerInfo.fStreamerInfo.fStreamers.fStreamerReserved.fClassTag = 0x80000000 | classTagOffset;
+   WriteKey(&streamerInfo, streamerInfo.GetSize(), fControlBlock->fHeader.GetSeekInfo(), 100, 1,
+            "TList", "StreamerInfo", "Doubly linked list");
+   fControlBlock->fHeader.SetNbytesInfo(fFilePos - fControlBlock->fHeader.GetSeekInfo());
+
+   // Reserve the space for the RNTuple record, which will be written on commit
+   fControlBlock->fSeekNTuple = fFilePos;
+   RTFKey keyRNTuple(fControlBlock->fSeekNTuple, 100, strRNTupleClass, strRNTupleName, strEmpty,
+                     fControlBlock->fNTuple.GetSize());
+
+   // The key index of the root TFile object, containing for the time being only the RNTuple key
+   fileRoot.fInfoShort.fSeekKeys = fFilePos + keyRNTuple.GetSize();
+   RTFKeyList keyList{1};
+   RTFKey keyKeyList(fileRoot.GetSeekKeys(), 100, strEmpty, strEmpty, strEmpty, keyList.GetSize() + keyRNTuple.fKeyLen);
+   Write(&keyKeyList, keyKeyList.fKeyHeaderSize, fileRoot.GetSeekKeys());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&keyList, keyList.GetSize());
+   Write(&keyRNTuple, keyRNTuple.fKeyHeaderSize);
+   Write(&strRNTupleClass, strRNTupleClass.GetSize());
+   Write(&strRNTupleName, strRNTupleName.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   fileRoot.fNBytesKeys = fFilePos - fileRoot.GetSeekKeys();
+
+   Write(&keyRoot, keyRoot.fKeyHeaderSize, 100);
+   Write(&strTFile, strTFile.GetSize());
+   Write(&strFileName, strFileName.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&strFileName, strFileName.GetSize());
+   Write(&strEmpty, strEmpty.GetSize());
+   Write(&fileRoot, fileRoot.GetSize());
+
+   fControlBlock->fNTuple.fSeekHeader = fileRoot.GetSeekKeys() + fileRoot.fNBytesKeys;
+}
+
+
+void ROOT::Experimental::Detail::RPageSinkFile::DoCreate(const RNTupleModel & /* model */)
```

Comment:
(I realize this predates this PR).  What is the semantic difference between ::Create and ::DoCreate? (i.e. I find 'Do' in function name 'often' either redundant or uninformative).

Reply:
The `DoXyz()` names are supposed to indicate a pattern where the base class has a public function `Xyz()` that does some common pre- and/or post-processing work around the actual logic.  The logic is implemented as virtual protected method `DoXyz()` by the derived classes.

Reply:
So Maybe 'XyzImpl'  or a name descriptive of the sub-part that needs to be specified by the sub-class?